### PR TITLE
Best hand completed, it is a lot more readable now! (Can be optimised further...)

### PR DESCRIPTION
**Pull request by xRayWx**
… messy code

### CODE REVIEW

File: src/assignments/ray/lesson_3_answers.clj [hunk 1]
```diff
@@ -48,7 +48,156 @@
...
+(= (best-hand-3 ["DQ" "HQ" "CA" "HA" "DJ"]) :two-pair)
+(= (best-hand-3 ["SQ" "DQ" "HQ" "C9" "DJ"]) :three-of-a-kind)
+(= (best-hand-3 ["SK" "DK" "HK" "CK" "DJ"]) :four-of-a-kind)
+(= (best-hand-3 ["SJ" "DJ" "HJ" "D9" "H9"]) :full-house)
+(= (best-hand-3 ["DQ" "DK" "DA" "D9" "DJ"]) :flush-or-straight-flush)
+
+;; Fourth iteration: All conditions mapped
+
+(defn subseq? [sub] 
+  (some #{sub} (partition 5 1 (map vector "A23456789TJQKA"))))
```
Comment (alain-picard) [inline src/assignments/ray/lesson_3_answers.clj]: Oooh!  I like that idea!  Nice use of partition!

File: src/assignments/ray/lesson_3_answers.clj [hunk 2]
```diff
@@ -48,7 +48,156 @@
...
+    (println count-suits)
+    (println "val-count-suits" val-count-suits)
+    (println count-suits-count)
+    (println count-ranks)
+    (println count-ranks-count)
+    (println "val-count-ranks" val-count-ranks)
+    (println "count-val-count-ranks" count-val-count-ranks)
+    (cond
+      ;; conditions explained, further differentiate will be done in the next iteration
+      (= count-ranks-count 5) :straight-flush
```
Comment (alain-picard) [inline src/assignments/ray/lesson_3_answers.clj]: I'm hopelessly lost in the counts and the ranks and the vals and the count-ranks and the rank-counts.

I'm afraid this code is pretty unreadable... :-(

Comment (xRayWx) [reply src/assignments/ray/lesson_3_answers.clj]: Yeah, sorry Alain, the prints were not meant to be readable, it was just used by me printing the result out for testing purposes. I realised this wasn't a great idea to be included in commits so I never include them in any of my subsequent commits.

File: src/assignments/ray/lesson_3_answers.clj [hunk 3]
```diff
@@ -48,8 +48,92 @@
...
+(defn best-hand-4 [hand]
+  ;; getting suits and ranks counts for initial hand determination
+  (let [[suit rank] (apply map vector hand)
+        count-suits (frequencies suit)
+        val-count-suits (count (frequencies suit))
+        count-suits-count (count count-suits)
+        count-ranks (frequencies rank)
+        count-ranks-count (count count-ranks)
+        val-count-ranks (vals (frequencies rank))
+        count-val-count-ranks (count (vals (frequencies rank)))
```
Comment (xRayWx) [inline src/assignments/ray/lesson_3_answers.clj:66]: Sorry, these are still here, as I said this will be further optimised

**Pull request merged**